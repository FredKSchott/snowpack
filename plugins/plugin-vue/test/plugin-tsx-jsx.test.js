const path = require('path');
const pluginTsxJsx = require('../plugin-tsx-jsx.js');

test('plugin with tsx', async () => {
  const pluginInstance = pluginTsxJsx({
    buildOptions: {
      sourceMap: true,
    },
  });
  const pluginLoad = pluginInstance.load;
  const codeFilePath = path.resolve(__dirname, './stubs/TsxContent.tsx');
  const resultContent = await pluginLoad({
    filePath: codeFilePath,
    fileExt: '.tsx',
  });
  expect(resultContent).toMatchSnapshot();
});

test('plugin with jsx', async () => {
  const pluginInstance = pluginTsxJsx({
    buildOptions: {
      sourceMap: true,
    },
  });
  const pluginLoad = pluginInstance.load;
  const codeFilePath = path.resolve(__dirname, './stubs/JsxContent.jsx');
  const resultContent = await pluginLoad({
    filePath: codeFilePath,
    fileExt: '.jsx',
  });
  expect(resultContent).toMatchSnapshot();
});
