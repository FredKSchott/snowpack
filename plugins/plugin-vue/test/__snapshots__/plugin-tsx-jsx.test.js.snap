// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin with jsx 1`] = `
"import { Fragment } from '/web_modules/vue.js';
import {createVNode, isVNode} from '/web_modules/vue.js';
const slice = Array.prototype.slice;
export function jsx(tag, props = null, children = null) {
  if (arguments.length > 3 || isVNode(children)) {
    children = slice.call(arguments, 2);
  }
  return createVNode(tag, props, children);
}

import {defineComponent} from \\"vue\\";
export default defineComponent({
  name: \\"JsxContent\\",
  setup() {
    return () => /* @__PURE__ */ jsx(Fragment, null, /* @__PURE__ */ jsx(\\"h1\\", null, \\"JsxContent\\"));
  }
});"
`;

exports[`plugin with tsx 1`] = `
"import { Fragment } from '/web_modules/vue.js';
import {createVNode, isVNode} from '/web_modules/vue.js';
const slice = Array.prototype.slice;
export function jsx(tag, props = null, children = null) {
  if (arguments.length > 3 || isVNode(children)) {
    children = slice.call(arguments, 2);
  }
  return createVNode(tag, props, children);
}

import {defineComponent} from \\"vue\\";
export default defineComponent({
  name: \\"TsxContent\\",
  setup() {
    const name = \\"TsxContent\\";
    return () => /* @__PURE__ */ jsx(Fragment, null, /* @__PURE__ */ jsx(\\"h1\\", null, \\"TsxContent\\"));
  }
});"
`;
